docs/TASK.md

```diff
@@ -48,87 +48,102 @@ I am a user who wants to define my custom task and bring my model to Disco, with
 
 ## 3) Procedure for adding a custom task
 
-In order to add a new custom task to Disco.js, we first need to create and export a new instance of the `Task` class, defined [here](../discojs/discojs-core/src/task/task.ts).
-Then, we must also export a function called `model` that specifies a model architecture for the task.
-We have to remember to export the task and the function in the `index.ts` [file which lives in the same folder](../discojs/discojs-core/src/tasks/index.ts).
-Finally, we need to rebuild Disco.js: `cd discojs/ && npm run build`
-
-
+In order to add a new custom task to Disco.js, we need to have defined a `TaskProvider` which need to implement two methods:
+   * `getTask` which returns a `Task` as defined [here](../discojs/discojs-core/src/task/task.ts), the `Task` contains all the crucial information from training to the mode
+   * `getModel` which returns a `Promise<tf.LayersModel>` specifying a model architecture for the task
 
+You can find examples of `TaskProvider` currently used in our Disco server in `discojs/discojs-core/src/default_tasks/`. These tasks are all loaded by our server by default.
 
 ### Task
 
 For the task creation, we consider the main use case which does not go through the user interface : 
 
 **I am a developper who wants to define my own task**
 
-In this case, your model and task will be uploaded and stored on our DISCO servers. You will have to make the task visible to the API. For your custom model, the JSON model architecture is necessary, but the .bin weight file is optional : if you include the weights file, your model will be loaded with the passed weights. If a weights file is not specified, the weights for the model will be initialized randomly.
+If you want to add a new task to our production DISCO server you have two possibilities:
+  * using the user interface as described above
+  * exporting your own `TaskProvider` from `discojs/discojs-core/src/default_tasks/`  and adding a new default task by contributing to the code.
```

Comment:
maybe say first one is without any coding required, second one is with describing the ML task in your own code